lib/matplotlib/tests/test_axes.py

```diff
@@ -4956,32 +4952,24 @@ def _helper_y(ax):
         r = ax.yaxis.get_major_locator()()
         assert r[-1] > 14
 
-    if request.param == 'x':
-        return _helper_x
-    elif request.param == 'y':
-        return _helper_y
-    else:
-        assert False, 'Request param %s is invalid.' % (request.param, )
+    return {"x": _helper_x, "y": _helper_y}[request.param]
 
 
 @pytest.fixture(params=['gca', 'subplots', 'subplots_shared', 'add_axes'])
 def shared_axes_generator(request):
     # test all of the ways to get fig/ax sets
-    if request.param == 'gca':
-        fig = plt.figure()
-        ax = fig.gca()
-    elif request.param == 'subplots':
-        fig, ax = plt.subplots()
-    elif request.param == 'subplots_shared':
-        fig, ax_lst = plt.subplots(2, 2, sharex='all', sharey='all')
-        ax = ax_lst[0][0]
-    elif request.param == 'add_axes':
-        fig = plt.figure()
-        ax = fig.add_axes([.1, .1, .8, .8])
-    else:
-        assert False, 'Request param %s is invalid.' % (request.param, )
-
-    return fig, ax
+    fig = plt.figure()
+    make_ax = {
+        "gca":
+            lambda fig: fig.gca(),
+        "subplots":
+            lambda fig: fig.subplots(),
+        "subplots_shared":
+            lambda fig: fig.subplots(2, 2, sharex="all", sharey="all")[0, 0],
+        "add_axes":
+            lambda fig: fig.add_axes([.1, .1, .8, .8]),
+    }[request.param]
+    return fig, make_ax(fig)
```

Comment:
I don't see this code prettier and actually it only partly checks what the functions return.

Reply:
The point is to not have to add code to handle a case that should never happen (request.param being an invalid value -- it is controlled by the fixture params) while keeping clear the fact that an error will be thrown if an invalid value ever comes in (... through some severely messed up bug in pytest).
If others also disagree (happy to hear more opinions), I'll revert.

Reply:
I'm not sure the use of `lambda`s stored in a `dict` is improving readability here. Removing the `else` on the if is fine with me.

Reply:
Removed the offending part.